tfjs-core/src/tensor_util_env.ts

```diff
@@ -18,15 +18,19 @@
 import {ENGINE} from './engine';
 import {env} from './environment';
 import {Tensor} from './tensor';
-import {DataType, TensorLike} from './types';
+import {DataType, TensorLike, WebGLData} from './types';
 import {assert, flatten, inferDtype, isTypedArray, toTypedArray} from './util';
 
-export function inferShape(val: TensorLike, dtype?: DataType): number[] {
+export function inferShape(
+    val: TensorLike|WebGLData, dtype?: DataType): number[] {
   let firstElem: typeof val = val;
 
   if (isTypedArray(val)) {
     return dtype === 'string' ? [] : [val.length];
   }
+  if (typeof val === 'object' && 'texture' in val) {
+    return [val.height, val.width];
```

Comment:
Great catch, done.

Reply:
Also, since we need to fill 'RGBA' as default if `channels` parameter is null. I moves this 'filling default' logics from backend to core, because to do this inferring shape in core, we also need to know the channels in core.